### PR TITLE
Chore: Remove global setting var from azure ad oath

### PR DESCRIPTION
**Pull request by idafurjes**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

### CODE REVIEW

File: pkg/login/social/social.go [hunk 1]
```diff
@@ -79,7 +79,7 @@ func newSocialBase(name string, config *oauth2.Config, info *setting.OAuthInfo)
 	}
 }
 
-func NewOAuthService() {
+func NewOAuthService(cfg *setting.Cfg) {
```
Comment (marefr) [inline pkg/login/social/social.go:82]: I see potential improvement of some of the other code here if you're up for it. Basically all references to reading settings using `settings.<setting>` could be replaced with using `cfg.<setting>` instead.

For example the line below `sec := setting.Raw.Section("auth." + name)` could be replaced with `sec := cfg.Raw.Section("auth." + name)`

**Review by marefr — COMMENTED** [review]
Looks good. But I see some additional improvements that can be made, see comment. 

In addition, the `setting.OAuthService` feels like a weird thing currently, but let's keep that for now and we could potentially getting rid of that in a separate PR.

File: pkg/login/social/social.go (hunk 1 above)
Comment (idafurjes) [reply pkg/login/social/social.go:82]: ok, will look into it

File: pkg/setting/setting.go [hunk 2]
```diff
@@ -385,6 +385,9 @@ type Cfg struct {
 	// Grafana Live ws endpoint (per Grafana server instance). 0 disables
 	// Live, -1 means unlimited connections.
 	LiveMaxConnections int
+
+	// Grafana.NET URL
```
Comment (marefr) [inline pkg/setting/setting.go:389]:
```suggestion
	// Grafana.com URL
```

File: pkg/login/social/social.go [hunk 3]
```diff
@@ -205,11 +205,11 @@ func NewOAuthService(cfg *setting.Cfg) {
...
 				ClientSecret: info.ClientSecret,
 				Endpoint: oauth2.Endpoint{
-					AuthURL:   setting.GrafanaComUrl + "/oauth2/authorize",
-					TokenURL:  setting.GrafanaComUrl + "/api/oauth2/token",
+					AuthURL:   cfg.GrafanaComURL + "/oauth2/authorize",
+					TokenURL:  cfg.GrafanaComURL + "/api/oauth2/token",
 					AuthStyle: oauth2.AuthStyleInHeader,
 				},
-				RedirectURL: strings.TrimSuffix(setting.AppUrl, "/") + SocialBaseUrl + name,
+				RedirectURL: strings.TrimSuffix(cfg.AppURL, "/") + SocialBaseUrl + name,
```
Comment (marefr) [inline pkg/login/social/social.go:212]: Usage of `setting.GrafanaComUrl` on line 218 below can be replaced with `cfg.GrafanaComURL`

**Review by marefr — CHANGES_REQUESTED** [review]
Looks good. Some minor changes left.

File: pkg/login/social/social.go (hunk 3 above)
Comment (idafurjes) [reply pkg/login/social/social.go:212]: ups, forgot to change it here

**Review by marefr — APPROVED** [review]
LGTM :rocket:

**Pull request merged**